app/views/admin/conferences/edit.html.haml

```diff
@@ -22,8 +22,12 @@
         = f.input :timezone, as: :time_zone, hint: 'The conference time zone'
         = f.input :start_date, as: :string, input_html: { id: 'conference-start-datepicker', readonly: 'readonly' }
         = f.input :end_date, as: :string, input_html: { id: 'conference-end-datepicker', readonly: 'readonly' }
-        = f.input :start_hour, input_html: {size: 2, type: 'number', min: 0, max: 23}
-        = f.input :end_hour, input_html: {size: 2, type: 'number', min: 1, max: 24}
+        - if @conference.program.events.scheduled(@conference.program.selected_schedule_id).count!=0
```

Comment:
Consider moving the count calculation back to the controller, e.g:
```
@affected_event_count = @conference.program.events.scheduled(@conference.program.selected_schedule_id).count
```

And adding a helper for the conditional hint:

```
# ApplicationHelper
def rescheduling_hint(affected_event_count)
  if affected_event_count > 0
    "You have #{affected_event_count} scheduled events. Changing the conference hours will unschedule those scheduled outside the conference hours."
  end
end
```

Then, all you need in the view is:

```
= f.input :start_hour, input_html: { size: 2, type: 'number', min: 0, max: 23 }, hint: rescheduling_hint(@affected_event_count)
= f.input :end_hour, input_html: { size: 2, type: 'number', min: 1, max: 24 }, hint: rescheduling_hint(@affected_event_count)
```

Less logic in the views is better.

Reply:
:smile: thanks